src/base/settingsstorage.h

```diff
@@ -62,4 +62,27 @@ public slots:
     mutable QReadWriteLock m_lock;
 };
 
+template<typename T>
+class StorableValue
```

Comment:
You're fixated on your current implementation. Look at the world wider. Why do you need setting a default value every time? Why do you need storing the default value in the config file?
Just do the following in TorrentCreatorDlg constructor initializer list:
```c++
, m_storeDialogSize(SETTINGS_KEY("Dimension"), size())
```